src/engine/enginebuffer.cpp

```diff
@@ -1163,12 +1163,24 @@ void EngineBuffer::processSeek(bool paused) {
             return;
     }
 
+    double syncPosition = position;
     if (!paused && (seekType & SEEK_PHASE)) {
-        position = m_pBpmControl->getNearestPositionInPhase(position, true, true);
+        syncPosition = m_pBpmControl->getNearestPositionInPhase(position, true, true);
     }
-    if (position != m_filepos_play) {
-        setNewPlaypos(position, adjustingPhase);
+
+    if (syncPosition != m_filepos_play) {
+        if(seekType & SEEK_PHASE){
+            // inform controllers about the seek position itself
+            // and do the phase shift afterwards
+            // this prevents loops from getting disabled because the synced position is in front of the loop
+            setNewPlaypos(position, false);
+            adjustingPhase = true;
+        }
+        setNewPlaypos(syncPosition, adjustingPhase);
     }
+
+    // seek finished after setNewPlaypos
+    m_iSeekQueued.fetchAndStoreRelaxed(SEEK_NONE);
```

Comment:
Are you able to justify that the reordering of atomic memory operations still works as expected?

Reply:
Actually, I couldn't. But it this line won't be necessary anymore.